test/README.md

```diff
@@ -218,3 +218,28 @@ Along with the environment variables provided by the Makefile (see above), an ad
 
     SOEXT              platform-specific extension for shared object files (aka. dynamic libraries),
                        e.g. .so, .dll or .dylib
+
+Test configuration variables
+----------------------------
+
+Sometimes test configuration arguments must be dynamic.
+For example, the output of all tests should be placed into `RESULTS_DIR`:
+
+```
+// REQUIRED_ARGS: -mixin=${RESULTS_DIR}/fail_compilation/mixin_test.mixin
+```
+
+Currently these variables are exposed:
+
+    RESULTS_DIR       Path to `test_results`
```

Comment:
I think I'm going to add at least `OUTPUT_BASE` as that's very handy too, but I will do this in another PR.
With `OUTPUT_BASE` it would just be:

```
REQUIRED_ARGS: -mixin=${OUTPUT_BASE}.mixin
```